trl/trainer/ppov2_bandit_rloo_trainer.py

```diff
@@ -0,0 +1,704 @@
+        if args.truncate_token and args.truncate_token == "eos":
+            args.truncate_token_id = tokenizer.eos_token_id
+        self.model = policy
+        self.create_optimizer_and_scheduler(num_training_steps=args.num_updates)
+
+        #########
+        ### trainer specifics
+        #########
+        self.state = TrainerState(
+            is_local_process_zero=self.is_local_process_zero(),
+            is_world_process_zero=self.is_world_process_zero(),
+        )
+        DEFAULT_CALLBACKS = [DefaultFlowCallback]
+        default_callbacks = DEFAULT_CALLBACKS + get_reporting_integration_callbacks(self.args.report_to)
+        if self.callbacks is None:
+            self.callbacks = default_callbacks
+        self.callback_handler = CallbackHandler(
+            self.callbacks, self.model, self.tokenizer, self.optimizer, self.lr_scheduler
+        )
+        self.control = TrainerControl()
+        self.is_deepspeed_enabled = getattr(self.accelerator.state, "deepspeed_plugin", None) is not None
+        self.is_fsdp_enabled = getattr(self.accelerator.state, "fsdp_plugin", None) is not None
+
+    # def get_train_dataloader(self) -> DataLoader:
+
+    def save_model(self, output_dir: Optional[str] = None):
+        """
+        Copied from Trainer.save_model, simplified.
+        By default we only save the policy and not the value network.
+        """
+        unwrapped: PreTrainedModel = self.accelerator.unwrap_model(self.model)
+        if output_dir is None:
+            output_dir = self.args.output_dir
+        state_dict = self.accelerator.get_state_dict(unwrapped)
+        if self.args.should_save:
+            self._save(output_dir, state_dict=state_dict)
+
+    def train(self):
+        args = self.args
+        accelerator = self.accelerator
+        local_seed = self.local_seed
+        optimizer = self.optimizer
+        model = self.model
+        ref_policy = self.ref_policy
+        reward_model = self.reward_model
+        tokenizer = self.tokenizer
+
+        dataloader = DataLoader(
+            self.train_dataset,
+            batch_size=args.local_batch_size,
+            shuffle=True,
+            collate_fn=DataCollatorWithPadding(tokenizer),
+            drop_last=True,  # needed; otherwise the last batch will be of ragged shape
+        )
+
+        device = accelerator.device
+        torch.backends.cudnn.deterministic = True # TODO: check if cudnn matters
+
+        # sync random states for DataLoader(shuffle=True) before `accelerator.prepare`
+        # see https://gist.github.com/vwxyzjn/2581bff1e48e185e0b85b6dfe1def79c
+        torch.manual_seed(args.seed)
+        model, optimizer, dataloader = accelerator.prepare(model, optimizer, dataloader)
+        torch.manual_seed(local_seed)  # reset the local seed again
+
+        def repeat_generator():
+            while True:
+                yield from dataloader
+
+        iter_dataloader = iter(repeat_generator())
+        if args.deepspeed2:
```

Comment:
You should be able to infer this from the `accelerator` directly - see how we did it for DPO

Reply:
Will check it out. TODO.